agent/logger/format.go

```diff
@@ -0,0 +1,87 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package logger
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"sync"
+)
+
+var (
+	formatPool *sync.Pool
+)
+
+type seelogMessageFormatter interface {
+	Format(message string, fields ...Fields) (string, error)
+}
+
+type messageJsonFormatter struct {
+}
+
+func (f *messageJsonFormatter) Format(message string, fields ...Fields) (string, error) {
+	buf := bufferPool.Get()
+	buf.Reset()
+	defer bufferPool.Put(buf)
+	enc := json.NewEncoder(buf)
+	data := make(map[string]interface{})
+	data["msg"] = message
+	for _, fi := range fields {
+		for k, v := range fi {
+			data[k] = v
+		}
+	}
+	if err := enc.Encode(data); err != nil {
+		return "", fmt.Errorf("failed to marshal message to JSON, %v", err)
+	}
+	return strings.TrimSpace(string(buf.Bytes())), nil
+}
+
+type messageTextFormatter struct {
+}
+
+func (f *messageTextFormatter) Format(message string, fields ...Fields) (string, error) {
+	buf := bufferPool.Get()
+	buf.Reset()
```

Comment:
maybe buf.Reset() could be done by the pool's Get() function

Reply:
I like it.